LayoutTests/fast/viewport/ios/content-visibility-layout-viewport-during-unstable-scroll.html

```diff
@@ -0,0 +1,157 @@
+        <section><p> </p></section>
+        <section><p> </p></section>
+        <section><p> </p></section>
+        <section><p> </p></section>
+        <section><p> </p></section>
+        <section><p> </p></section>
+        <section><p> </p></section>
+        <section><p> </p></section>
+        <section><p> </p></section>
+        <section><p> </p></section>
+        <section><p> </p></section>
+        <section><p> </p></section>
+        <section><p> </p></section>
+        <section><p> </p></section>
+        <section><p> </p></section>
+        <section><p> </p></section>
+        <section><p> </p></section>
+        <section><p> </p></section>
+        <section><p> </p></section>
+        <section><p> </p></section>
+        <section><p> </p></section>
+        <section><p> </p></section>
+        <section><p> </p></section>
+        <section><p> </p></section>
+        <section><p> </p></section>
+    </main>
+    <div id="console"></div>
+    <script src="../../../resources/js-test.js"></script>
+    <script src="../../../resources/ui-helper.js"></script>
+    <script>
+        jsTestIsAsync = true;
+
+        function nthSection(n) {
+            return document.querySelectorAll("p")[n - 1];
+        }
+
+        function numberOfVisibleItems() {
+            return [...document.querySelectorAll("p")]
+                .filter(element => element.checkVisibility({ contentVisibilityAuto: true }))
+                .length;
+        }
+
+        function scrollingStartScript(offsetTop) {
+            return `(function() {
+                uiController.stableStateOverride = false;
+                uiController.beginInteractiveObscuredInsetsChange();
+                uiController.immediateScrollToOffset(0, ${offsetTop});
+                uiController.doAfterPresentationUpdate(function() {
+                    uiController.doAfterPresentationUpdate(function() {
+                        uiController.doAfterPresentationUpdate(function() {
+                            uiController.uiScriptComplete();
+                        });
+                    });
+                });
+            })()`;
+        }
+        function scrollingEndScript() {
+            return `(function() {
+                uiController.stableStateOverride = true;
+                uiController.doAfterNextStablePresentationUpdate(function() {
+                    uiController.endInteractiveObscuredInsetsChange();
+                    uiController.uiScriptComplete();
+                });
+            })()`;
+        }
+
+        addEventListener("load", async () => {
+            await UIHelper.ensurePresentationUpdate();
+
+            const initialVisibleItems = numberOfVisibleItems();
+            shouldBeTrue(`nthSection(1).checkVisibility({ contentVisibilityAuto: true })`);
+            shouldBeTrue(`nthSection(2).checkVisibility({ contentVisibilityAuto: true })`);
+            testRunner.runUIScript(scrollingStartScript(300), function() {
```

Comment:
Can this be written without the nesting? Can the test be modified to take less than a second?